checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java

```diff
@@ -218,6 +218,13 @@ protected void constructorFromUsePreSubstitution(
    */
   private void changeNonOwningParameterTypesToTop(
       ExecutableElement declaration, AnnotatedExecutableType type) {
+    // Formal parameters without a declared owning annotation are disregarded by the RLC analysis,
+    // as their @MustCall obligation is set to Top in this method. However, this computation is not
+    // desirable for RLC inference in unannotated programs, where the goal is to infer and add
+    // @Owning annotations to owning parameters.
+    if (getWholeProgramInference() != null) {
```

Comment:
This test will be true when WPI is running (even if `MustCallInferenceLogic` is not), so I think this test should be more specific: probably `isWPIEnabledForRLC()` method you use in `CalledMethodsTransfer` or a similar mechanism.